app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,7 +67,17 @@ def pxe_with_building_hosts?(template)
       template.respond_to?(:operatingsystem_ids) &&
       template.template_kind.present? &&
       kinds.include?(template.template_kind.name) &&
-      Host.where(:build => true, :operatingsystem_id => template.operatingsystem_ids).any?
+      building_hosts(template).any?
+  end
+
+  def building_hosts(template)
+    Host.where(:build => true, :operatingsystem_id => template.operatingsystem_ids)
+  end
+
+  def link_to_selected_hosts(text, host_ids)
+    cookies.delete(:'_ForemanSelectedhosts') if cookies[:'_ForemanSelectedhosts']
+    cookies[:'_ForemanSelectedhosts'] = { :value => JSON.generate(host_ids), :secure => true }
```

Comment:
Couple of things here:

a) I don't think we should change the user's selected cookie just by viewing the provisioning template page, this could be very surprising and even dangerous.  I'd suggest either not trying to implement this functionality, or do it via the search function - such as adding a new (hidden) search term to hosts for `uses_template_id = 6` or something that's easy to link to.
b) secure => true only works when the server's HTTPS.